extensions/positron-catalog-explorer/src/catalogs/snowflake.ts

```diff
@@ -73,195 +65,223 @@ export function registerSnowflakeProvider(
 }
 
 /**
- * Register a Snowflake catalog provider using SSO authentication.
+ * Register a Snowflake catalog provider using authentication details from connections.toml.
  */
 export async function registerSnowflakeCatalog(
 	context: vscode.ExtensionContext,
-	account?: string,
+	connOptions?: string | any,
+	connName?: string
 ): Promise<CatalogProvider | undefined> {
-	try {
-		// Get recent account names to suggest to the user
-		const recentAccounts = context.globalState.get<string[]>(RECENT_SNOWFLAKE_ACCOUNTS_KEY) || [];
-
-		// User is not re-authenticating an existing account, prompt for account name
-		if (!account) {
-			// If we have recent accounts, show a quick pick first
-			if (recentAccounts.length > 0) {
-				// Add a "New Account" option at the end
-				const items = [
-					...recentAccounts.map(name => ({
-						label: name,
-						description: 'Recent account'
-					})),
-					{
-						label: 'Enter New Account...',
-						description: 'Provide a different account identifier'
-					}
-				];
-
-				const selection = await vscode.window.showQuickPick(items, {
-					placeHolder: 'Select a recent account or enter a new one',
-					ignoreFocusOut: true
-				});
-
-				if (!selection) {
-					return undefined; // User canceled
+	// Load all connections from connections.toml
+	const connections = await getSnowflakeConnectionOptions();
+
+	// User is not re-authenticating an existing connection, prompt for one
+	if (!connName) {
+		let items: vscode.QuickPickItem[] = [];
+
+		if (connections && Object.keys(connections).length > 0) {
+			// If we found connections, add them to the quick pick
+			const connectionNames = Object.keys(connections);
+			items = connectionNames.map(name => ({
+				label: name,
+				description: `Connection from account (${connections[name].account || 'No account specified'})`
+			}));
+		} else {
+			// If no connections found, offer help options
+			items = [
+				{
+					label: 'No connections.toml file found',
+					description: 'Configure where to look for connections.toml',
+					detail: 'Create a connections.toml file in ~/.snowflake/ or set a custom path in settings.'
```

Comment:
Is selecting this option meant to do anything? This reads to me like when I select it, it will help me configure a toml file. But now when I select it, I just get a 'cannot read properties of undefined' toast.